test/e2e/resource_validation_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+package e2e
```

Comment:
please move this to integration test

Reply:
hi ryan, will the e2e tests be better here? The invalid MC is rejected by the CRD definition, while "The [controller-runtime/pkg/envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest) Go library helps write integration tests for your controllers by setting up and starting an instance of etcd and the Kubernetes API server, without kubelet, controller-manager or other components."

The ITs is better to test the controller behavior itself.

Reply:
The test here is just to get Api-server behavior so the [controller-runtime/pkg/envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest) is a perfect place to test this feature since setup a real API-server in the env. Arvind already have a few IT test to do that.

Reply:
please refer to this existing integration tests. https://github.com/Azure/fleet/blob/main/test/apis/placement/v1alpha1/api_validation_integration_test.go